Add a new material for evaluating material properties on a control drum (#32773)
idaholab/moose: modules/reactor/doc/content/source/materials/DrumMaterial.md

> # DrumMaterial

Reviewer: ```suggestion
# ControlDrumMaterial
```
all the other objects dont call it just Drum

!syntax description /Materials/DrumMaterial

## Description

> `DrumMaterial` supports evaluations of a material property with rotations.

Reviewer: ```suggestion
`DrumMaterial` supports evaluations of a material property with the rotation corresponding to the current position of the control drum(s).
```

The rotating component that an mesh element belongs to has the minimum distance between its rotation center and the average of element vertices.
The rotation components may have multiple *segements* rotation-wise in [!param](/Materials/DrumMaterial/segment_angles), although all components must have the same number of segments.
Values in [!param](/Materials/DrumMaterial/segment_angles) must sum to 360 degree to cover the entire rotation range.
Material property, whose name is specified with [!param](/Materials/DrumMaterial/drum_material_property), are evaluated from coupled material properties of all segments in [!param](/Materials/DrumMaterial/segment_material_properties).
The rotation positions of all rotating components, or more precisely the rotation angle of the starting edge of the first segment, are specified by functions in [!param](/Materials/DrumMaterial/rotation_angle_functions) and [!param](/Materials/DrumMaterial/rotation_angle_offsets).
The size of [!param](/Materials/DrumMaterial/rotation_angle_functions) can be equal to the number of rotating components or one when all components share the same function.

> However, the size of [!param](/Materials/DrumMaterial/rotation_angle_offsets) must be equal to the number of rotating components.

Reviewer: this paragraph is messy, can you please separate the rotation parameters from discretization from location from properties

Finally, it determines which segment this quadrature point belongs to based on the angle, the current rotation position by [!param](/Materials/DrumMaterial/rotation_angle_functions) and [!param](/Materials/DrumMaterial/rotation_angle_offsets), and assigns the drum material property with the segment material property in [!param](/Materials/DrumMaterial/segment_material_properties).

!alert note
There could be situations quadrature points within a single element belongs to different segments.
Material property values are different on the quadrature points.
The integration of the material property on the element divided by element volume is an approximation of the volume homogenized property.
These typically do not introduce significant discretization errors for thermal conduction, but does require some special treatment for neutroncs often known as decusping.

!alert note
The material only evaluate the material property according to the rotation position.
It does not consider the physics due to the component rotation during transient.

> A convection term with component rotition speed needs to be introduced for completeness, but this physics can often be neglected in modeling efforts.

Reviewer: ```suggestion
A convection term with component rotiaion speed needs to be introduced for completeness, but this physics can often be neglected in modeling efforts.
```
